-- GunPrices.luau
-- ModuleScript for storing and retrieving gun prices by name

local GunPrices = {
    ["AK-105"] = 2500,
    ["AK-47"] = 2000,
    ["AK-47 DORADA"] = 5000,
    ["AR15"] = 2200,
    ["AS VAL"] = 3500,
    ["AUG A2"] = 2700,
    ["AmmoBox"] = 100,
    ["Bat"] = 150,
    ["Baton"] = 120,
    ["Beanbag Shotgun"] = 1800,
    ["COLT XM177"] = 2100,
    ["CZ-805"] = 2600,
    ["Crowbar"] = 200,
    ["DEAGLE"] = 1800,
    ["Desert Deagle"] = 2200,
    ["Double Barrel Shotgun"] = 2400,
    ["Escudo"] = 800,
    ["Escudo Antibalas"] = 1200,
    ["FN FAL"] = 3200,
    ["FX-05"] = 2800,
    ["Famas"] = 2500,
    ["FiveSeven"] = 900,
    ["Flamethrower"] = 4000,
    ["Flashbang"] = 300,
    ["G3"] = 3100,
    ["Gas Mask"] = 500,
    ["Glock-17"] = 700,
    ["Glock-19"] = 750,
    ["Grenade"] = 400,
    ["Grenade Launcher"] = 5000,
    ["HK416"] = 3300,
    ["Handgun"] = 600,
    ["Hunting Rifle"] = 2000,
    ["K2"] = 2600,
    ["Kar98k"] = 2100,
    ["Laser Rifle"] = 6000,
    ["Lewis Gun"] = 3500,
    ["Luger"] = 800,
    ["M1 Garand"] = 2300,
    ["M1014"] = 2500,
    ["M110"] = 3200,
    ["M14"] = 2400,
    ["M16A2"] = 2200,
    ["M16A4"] = 2300,
    ["M249"] = 4200,
    ["M4 Carbine"] = 2700,
    ["M4A1"] = 2900,
    ["M4A1 CQBR"] = 2950,
    ["M4A1 Colt"] = 3000,
    ["M4A1 UGRI"] = 3100,
    ["M60"] = 4500,
    ["M82 Barrett"] = 7000,
    ["M9 Bayonet"] = 900,
    ["MCX"] = 3400,
    ["MG42"] = 4300,
    ["MP40"] = 1700,
    ["MP5"] = 1800,
    ["MP5K"] = 1850,
    ["MP7"] = 1900,
    ["Machete"] = 250,
    ["Magpul"] = 2100,
    ["Makarov"] = 600,
    ["Military Helmet"] = 400,
    ["Military Vest"] = 800,
    ["P90"] = 2000,
    ["PP-19"] = 2100,
    ["Pipe Bomb"] = 350,
    ["Pump Shotgun"] = 1600,
    ["RK62"] = 2300,
    ["RPK16"] = 4100,
    ["RailGun"] = 8000,
    ["Remington 870"] = 1700,
    ["Revolver"] = 1200,
    ["Riot Helmet"] = 450,
    ["Rocket Launcher"] = 6000,
    ["SCAR-H"] = 3500,
    ["SIG MCX"] = 3400,
    ["SIG-516"] = 3200,
    ["Sawed Off"] = 1400,
    ["Shotgun"] = 1500,
    ["Silenced Pistol"] = 950,
    ["Skorpion"] = 1100,
    ["Sledge Hammer"] = 300,
    ["Smoke Grenade"] = 200,
    ["Sniper Rifle"] = 3500,
    ["Spas-12"] = 2600,
    ["Sten Gun"] = 1700,
    ["Tabla"] = 100,
    ["Tactical Knife"] = 250,
    ["Taser"] = 500,
    ["Tec-9"] = 1000,
    ["Tommy Gun"] = 2100,
    ["USP"] = 850,
    ["UZI"] = 1200,
    ["Vector"] = 2000,
    ["X16"] = 700,
    ["AA-12"] = 5000
}

function GunPrices.GetPrice(name)
    return GunPrices[name] or 0
end

function GunPrices.GetAll()
    return GunPrices
end

return GunPrices
